Code/Framework/AzCore/AzCore/std/string/conversions.h

```diff
@@ -30,22 +30,43 @@ namespace AZStd
         template<size_t Size = sizeof(wchar_t)>
         struct WCharTPlatformConverter
         {
+            static_assert(Size == size_t{ 2 } || Size == size_t{ 4 }, "only wchar_t types of size 2 or 4 can be converted to utf8");
+
             template<class Allocator1>
             static inline void to_string(AZStd::basic_string<string::value_type, string::traits_type, Allocator1>& dest, const wchar_t* first, const wchar_t* last)
             {
                 if constexpr (Size == 2)
                 {
-                    Utf8::Unchecked::utf16to8(first, last, AZStd::back_inserter(dest));
+                    Utf8::Unchecked::utf16to8(first, last, AZStd::back_inserter(dest), dest.max_size());
                 }
                 else if constexpr (Size == 4)
                 {
-                    Utf8::Unchecked::utf32to8(first, last, AZStd::back_inserter(dest));
+                    Utf8::Unchecked::utf32to8(first, last, AZStd::back_inserter(dest), dest.max_size());
+                }
+            }
+
+            template<size_t MaxElementCount>
+            static inline void to_string(AZStd::basic_fixed_string<string::value_type, MaxElementCount, string::traits_type>& dest, const wchar_t* first, const wchar_t* last)
```

Comment:
The `value_type` and `traits_type` needs to be a template parameter.
It can't be hardcoded to be the `AZStd::string` value_type(`char`) and traits_type(`char_traits<char>`) as those aliases are separate from the basic_fixed_string class

Reply:
they need to be hardcoded because `to_string` cannot receive a `wchar_t` as the destination, for that there is a `to_wstring`
Note that the same is done for `basic_string`